nemo/collections/asr/parts/utils/speaker_utils.py

```diff
@@ -611,6 +613,29 @@ def isOverlap(rangeA, rangeB):
     return end1 > start2 and end2 > start1
 
 
+def validate_vad_manifest(AUDIO_RTTM_MAP, vad_manifest):
+    """
+    This function will validate the manifest file from NeMo voice activity detection or Oracle VAD to the uniq_ids present 
+    in provided audio manifest file. If there is a mis-match we ignore those uniq_ids from clustering.
+    """
+    vad_uniq_ids = set()
+    with open(vad_manifest, 'r') as vad_file:
+        for line in vad_file:
+            line = line.strip()
+            dic = json.loads(line)
+            if dic['duration'] > 0:
+                vad_uniq_ids.add(dic['uniq_id'])
+
+    provided_uniq_ids = set(AUDIO_RTTM_MAP.keys())
+    silence_ids = provided_uniq_ids - vad_uniq_ids
+    for uniq_id in silence_ids:
+        del AUDIO_RTTM_MAP[uniq_id]
+        logging.warning(f"{uniq_id} ignored due to silence detected in the session.")
```

Comment:
"silence detected" can be misleading to users since what we mean is there is no signal to be processed.
also, it should be passive voice (ignored -> is ignored)
It should be this way:
`f"{uniq_id} is ignored since the file does not contain any speech signal to be processed."`